pkg/cmd/grafana/apiserver/cmd.go

```diff
@@ -116,16 +117,17 @@ type lateInitializedTracingProvider struct {
 }
 
 func (tp lateInitializedTracingProvider) Tracer(name string, options ...trace.TracerOption) trace.Tracer {
-	return tp.tracer
+	return tp.tracer.getTracer()
 }
 
 type lateInitializedTracingService struct {
 	tracing.Tracer
+	mutex sync.RWMutex
 }
 
 func newLateInitializedTracingService() *lateInitializedTracingService {
 	ts := &lateInitializedTracingService{
-		Tracer: tracing.InitializeTracerForTest(),
+		Tracer: tracing.NewNoopTracerService(),
```

Comment:
Figured better to use a no op tracer instead of the in-memory one from `tracing.InitializeTracerForTest` when tracing is disabled.